internal/validation/duplicates.go

```
@@ -0,0 +1,237 @@
+}
+
+func FindAndRemoveDuplicates(conn clickhouse.Conn, chainId *big.Int, startBlock *big.Int, endBlock *big.Int) error {
+	duplicateBlockNumbers, err := findDuplicateBlocksInRange(conn, chainId, startBlock, endBlock)
+	if err != nil {
+		return err
+	}
+	if len(duplicateBlockNumbers) == 0 {
+		log.Debug().Msg("No duplicate blocks found in range")
+	} else {
+		log.Debug().Msgf("Found %d duplicate blocks in range %v-%v: %v", len(duplicateBlockNumbers), startBlock, endBlock, duplicateBlockNumbers)
+		removeDuplicateBlocks(conn, chainId, duplicateBlockNumbers)
```

golangci-lint (1.64.8)

Error return value is not checked

(errcheck)